src/Stratis.Bitcoin/Connection/ConnectionManagerBehavior.cs

```diff
@@ -99,7 +99,9 @@ private async Task OnStateChangedAsync(INetworkPeer peer, NetworkPeerState oldSt
         protected override void DetachCore()
         {
             this.AttachedPeer.StateChanged.Unregister(this.OnStateChangedAsync);
-            this.connectionManager.PeerDisconnected(this.AttachedPeer.Connection.Id);
+
+            if (this.AttachedPeer.Connection != null)
```

Comment:
`this.AttachedPeer` could also be null?

Reply:
I don't think so, see my comment from the PR description

Reply:
👍